docs/source/Howtos/Extending-the-REST-API.md

```diff
@@ -0,0 +1,287 @@
+# Extending the REST API
+
+*Note: This tutorial assumes you have a basic understanding of the `web` folder structure. If you don't, please read [the page on the website](https://www.evennia.com/docs/1.0-dev/Components/Website.html) and make sure you understand it before you come back!*
+
+For this tutorial, we'll be adding an `inventory` action to the `characters` endpoint, showing all objects being _worn_ and _carried_ by a character. The first thing you should do is review the [REST API](https://www.evennia.com/docs/1.0-dev/Components/Web-API.html) documentation page. It's not very long, but it covers how to turn the API on and which parts of Django you should be familiar with.
+
+Once you've read that and visited `/api/characters/`, it's time to get started!
+
+## Creating your own viewset
+
+The first thing you'll need to do is create your own views module: `<mygame>/web/api/views.py`
+
+The default REST API endpoints are controlled by classes in `evennia/web/api/views.py` - you could copy that entire file directly and use it, but we're going to focus on changing the minimum.
+
+To start, we'll reimplement the existing `characters` endpoint: a child view of the `objects` endpoint that can only access characters.
+
+```python
+"""
+<mygame>/web/api/views.py
+
+Customized views for the REST API
+"""
+# we'll need these from django's rest framework to make our view work
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework import status
+
+# this implements all the basic Evennia Object endpoint logic, so we're inheriting from it
+from evennia.web.api.views import ObjectDBViewSet
+
+# and we need this to filter our character view
+from evennia.objects.objects import DefaultCharacter
+
+# our own custom view
+class CharacterViewSet(ObjectDBViewSet):
+	"""
+	A customized Character view that adds an inventory detail
+	"""
+	queryset = DefaultCharacter.objects.all_family()
+```
+
+## Setting up the urls
+
+Now that we have a viewset of our own, we can create our own urls module and change the `characters` endpoint path to point to ours. The API routing is more complicated than the other `urls.py` files, so you need to copy evennia's from `evennia/web/api/urls.py` to your folder, `<mygame>/web/api/urls.py` and open it in your editor.
```

Comment:
Worth repeating what the `urls.py` module actually does - routing the URL onto a python view. Also an idea to link to the corresponding Django docs for those that want to dive deeper.